src/components/snapshot/AuthDomain.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { Box, FormLabel, Link, styled } from '@mui/material';
+import { ManagedGroupMembershipEntry } from 'src/models/group';
+import { AppDispatch } from 'src/store';
+import { TdrState } from 'src/reducers';
+import { useOnMount } from 'src/libs/utils';
+import { LaunchOutlined } from '@mui/icons-material';
+import { connect } from 'react-redux';
+import { getUserGroups } from 'src/actions';
+import JadeDropdown from '../dataset/data/JadeDropdown';
+
+const JadeLink = styled('span')(({ theme }) => theme.mixins.jadeLink);
+
+type AuthDomainProps = {
+  dispatch: AppDispatch;
+  userGroups: Array<ManagedGroupMembershipEntry>;
+  setParentAuthDomain: (domain: string) => void;
+};
+
+function AuthDomain({ dispatch, userGroups, setParentAuthDomain }: AuthDomainProps) {
+  const [selectedAuthDomain, setSelectedAuthDomain] = React.useState<string | undefined>(undefined);
+
+  useOnMount(() => {
+    dispatch(getUserGroups());
+  });
+
+  return (
+    <>
+      <FormLabel sx={{ fontWeight: 700, color: '#333f52' }} htmlFor="authorization-domain">
+        Authorization Domain
+        <span style={{ fontWeight: 400, fontStyle: 'italic' }}> - (optional)</span>
```

Comment:
This line also triggers a sonar warning, the idea being that a reader might think that the newline here is preserved. Their fix doesn't seem great to me though so I'd probably skip this one.